### ISSUE TITLE
Bug:NM attribute not cleared

### ISSUE THREAD

**Issue by fohlsso2** [cgeo/cgeo#4576]
If a cache do not have any attributes except the Need Maintenence flag, that attribute is not removed on refresh after the attribute is cleared on the geocaching site.

#3370 is closed but the bug is still there.

**Comment by samueltardieu** [comment]
Interestingly, we had two issues there. I've fixed the first one in the `release` branch, which should be enough to make the bug disappear, and the second one also in the `master` branch:

1. If no attributes section was found on the GC HTML page for the cache, it was not initialized at all in the cache, thus leading to a lazy-loading of the previous content from the database when accessed. It should have been initialized to an empty list to reflect the fact that it has indeed come from the parsing. Fixed by ddd07e93ecfa89e5d67da55161761393348d4c48.

2. Lazy-loading of some fields (attributes, logs, and some texts) happened irregardless of whether the cache originally came from the database or not. Fixed by 3085cb1bf05fc617a3225edd142b31d48314bff5.
